### PR TITLE
fix home attende

### PR DESCRIPTION
**Pull request by AbdeleahmanEl-Desoky**
## 🎯 Summary

-
## 🧾 Related Ticket

- Jira: [ABC-123](https://jira.your-company.com/browse/ABC-123)

## 🔄 Type of Change

- [ ] Feature
- [ ] Bugfix
- [ ] Refactor
- [ ] Hotfix
- [ ] Chore / Maintenance

## 🧪 How to Test

1. Call endpoint: `GET /api/...`
2. Use test data: `...`
3. Expected result:
   - 
    -

## ✅ Checklist

- [ ] Performed **self-review** of the code
- [ ] Updated **validations** if business logic changed
- [ ] Updated **API docs / Postman collection** if the endpoint changed
- [ ] All **tests are passing** (if applicable)
- [ ] No **breaking changes** for existing clients

## 🧱 Migration / Database Changes

- [ ] No database changes
- [ ] There is a new migration (describe below):

### CODE REVIEW

File: modules/Attendance/Services/UserAttendanceService.php [hunk 1]
```diff
@@ -828,7 +1121,9 @@ private function buildPeriodsFromAttendances(Collection $attendances): array
 
         return $groups
             ->sortBy(fn (Collection $group) => $this->earliestClockInTimestampInGroup($group))
-            ->map(fn (Collection $group) => $this->buildAggregatedPeriodFromShiftGroup($group))
+            ->map(function (Collection $group) {
+                return $this->stripInternalHistoryPeriodKeys($this->buildAggregatedPeriodFromShiftGroup($group));
```
Comment (AmrSaleh2234) [inline modules/Attendance/Services/UserAttendanceService.php:1125]: The method `stripInternalHistoryPeriodKeys` is called without any error handling or validation. Ensure this method exists and handles cases where the input might be invalid or null.

Comment (AmrSaleh2234) [inline modules/Attendance/Services/UserAttendanceService.php:1125]: Consider using arrow function syntax (`fn() =>`) instead of the full closure syntax for consistency with the rest of the codebase, unless there's a specific reason to use the full closure syntax.

Comment (AmrSaleh2234) [inline modules/Attendance/Services/UserAttendanceService.php:1125]: The nesting of method calls (`stripInternalHistoryPeriodKeys` inside `buildAggregatedPeriodFromShiftGroup`) makes the code harder to read. Consider assigning the intermediate result to a variable with a descriptive name before passing it to `stripInternalHistoryPeriodKeys`.

Comment (AmrSaleh2234) [inline modules/Attendance/Services/UserAttendanceService.php:1125]: If `stripInternalHistoryPeriodKeys` modifies the structure of the period data, ensure this change is properly documented in the method's return type hint or PHPDoc.

Comment (AmrSaleh2234) [inline modules/Attendance/Services/UserAttendanceService.php:1125]: The purpose of stripping internal history period keys isn't clear from this diff. Consider adding a comment explaining why this transformation is needed if it's not obvious from the method name.

Comment (AmrSaleh2234) [inline modules/Attendance/Services/UserAttendanceService.php:1125]: If `stripInternalHistoryPeriodKeys` could potentially throw exceptions, they should be caught and handled appropriately in this context.

Comment (AmrSaleh2234) [inline modules/Attendance/Services/UserAttendanceService.php:1125]: Ensure `stripInternalHistoryPeriodKeys` doesn't modify the original `$group` collection if that would cause unintended side effects.

Comment (AmrSaleh2234) [inline modules/Attendance/Services/UserAttendanceService.php:1125]: The performance impact of adding `stripInternalHistoryPeriodKeys` should be considered, especially if this code runs in a loop or performance-critical section.

Comment (AmrSaleh2234) [inline modules/Attendance/Services/UserAttendanceService.php:1125]: If `stripInternalHistoryPeriodKeys` filters out certain keys, make sure those keys aren't needed by any downstream code that consumes the result of `buildPeriodsFromAttendances`.

Comment (AmrSaleh2234) [inline modules/Attendance/Services/UserAttendanceService.php:1125]: Consider whether `stripInternalHistoryPeriodKeys` should be called inside `buildAggregatedPeriodFromShiftGroup` instead, if it's always needed after building an aggregated period.

Comment (AmrSaleh2234) [inline modules/Attendance/Services/UserAttendanceService.php:1125]: If `stripInternalHistoryPeriodKeys` is a new method, ensure it's properly tested with unit tests covering all edge cases.

Comment (AmrSaleh2234) [inline modules/Attendance/Services/UserAttendanceService.php:1125]: The change introduces a new dependency (`stripInternalHistoryPeriodKeys`) on the output format of `buildAggregatedPeriodFromShiftGroup`. Ensure this coupling is intentional and well-documented.

Comment (AmrSaleh2234) [inline modules/Attendance/Services/UserAttendanceService.php:1125]: Consider whether `stripInternalHistoryPeriodKeys` could be implemented as a method on the period object/array itself for better encapsulation.

Comment (AmrSaleh2234) [inline modules/Attendance/Services/UserAttendanceService.php:1125]: If `stripInternalHistoryPeriodKeys` removes sensitive data, ensure this is properly reflected in any API documentation that might expose this data structure.

Comment (AmrSaleh2234) [inline modules/Attendance/Services/UserAttendanceService.php:1125]: The diff doesn't show the implementation of `stripInternalHistoryPeriodKeys`. Reviewers should see this method to properly evaluate the change.
